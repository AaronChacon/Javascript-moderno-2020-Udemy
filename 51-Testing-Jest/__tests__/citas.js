import Citas from '../js/classes/Citas.js';

describe('Â´Probar la clase de citas', () =>{
    
    const citas = new Citas();

    const id = Date.now();

    test('Agregar una nueva cita', () => {
        
        const citaObj = {
            mascota: 'Coco',
            propietario: 'Aaron',
            telefono: '920889432',
            fecha: '11-03-2020',
            hora: '10:30',
            sintomas: 'No come',
            id
        }

        citas.agregarCita(citaObj);

        // Prueba

        expect(citas).toMatchSnapshot();
        
    });

    test('Actualizar cita', () => {

        const citaActualizada = {
            mascota: 'Coco 2',
            propietario: 'Aaron',
            telefono: '920889432',
            fecha: '11-03-2020',
            hora: '10:30',
            sintomas: 'No come',
            id
        }

        citas.editarCita(citaActualizada);

        // Prueba
        expect(citas).toMatchSnapshot();

    })
    
    test('Eliminar cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    })
    


    
});