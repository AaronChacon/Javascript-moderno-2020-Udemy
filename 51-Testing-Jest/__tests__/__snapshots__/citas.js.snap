// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`´Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "11-03-2020",
      "hora": "10:30",
      "id": 1604401797128,
      "mascota": "Coco 2",
      "propietario": "Aaron",
      "sintomas": "No come",
      "telefono": "920889432",
    },
  ],
}
`;

exports[`´Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "11-03-2020",
      "hora": "10:30",
      "id": 1604401797128,
      "mascota": "Coco",
      "propietario": "Aaron",
      "sintomas": "No come",
      "telefono": "920889432",
    },
  ],
}
`;

exports[`´Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
